apps/studio/components/interfaces/Account/NewAccessTokenButton.tsx

```diff
@@ -1,158 +1,185 @@
-                  </div>
-                </DropdownMenuItem>
-              </DropdownMenuContent>
-            </DropdownMenu>
-          </div>
-        </div>
-      </div>
-
-      <Modal
-        hideFooter
-        size="small"
-        visible={isOpen}
-        onCancel={() => setIsOpen(!isOpen)}
-        header={
-          <div className="flex items-baseline gap-2">
-            <h5 className="text-sm text-foreground">
-              {tokenScope === 'V0' ? 'Generate token for experimental API' : 'Generate New Token'}
-            </h5>
-          </div>
+    <Dialog
+      open={visible}
+      onOpenChange={(open) => {
+        if (!open) {
+          form.reset()
         }
-      >
-        <Form
-          validateOnBlur
-          initialValues={{ tokenName: '' }}
-          onSubmit={onFormSubmit}
-          validate={validate}
-        >
-          {() => (
-            <>
-              {tokenScope === 'V0' && (
-                <Modal.Content>
-                  <Admonition
-                    type="warning"
-                    title="The experimental API provides additional endpoints which allows you to manage
-                      your organizations and projects."
-                    description={
-                      <>
-                        <p>
-                          These include deleting organizations and projects which cannot be undone.
-                          As such, be very careful when using this API.
-                        </p>
-                        <div className="mt-4">
-                          <Button asChild type="default" icon={<ExternalLink />}>
-                            <Link
-                              href="https://api.supabase.com/api/v0"
-                              target="_blank"
-                              rel="noreferrer"
-                            >
-                              Experimental API documentation
-                            </Link>
-                          </Button>
-                        </div>
-                      </>
-                    }
-                  />
-                </Modal.Content>
-              )}
-              <Modal.Content>
-                <Input
-                  id="tokenName"
-                  label="Name"
-                  placeholder="Provide a name for your token"
-                  className="w-full"
-                />
-              </Modal.Content>
-              <DialogFooter>
-                <div className="flex items-center space-x-2 justify-end">
-                  <Button type="default" disabled={isLoading} onClick={() => setIsOpen(false)}>
-                    Cancel
-                  </Button>
-                  <Button htmlType="submit" loading={isLoading} disabled={isLoading}>
-                    Generate token
-                  </Button>
+        setVisible(open)
+      }}
+    >
+      <DialogTrigger asChild>
```

Comment:
DialogTrigger should only have one child

Reply:
Yeah we if we're controlling the Dialog with an external state this way, i reckon the buttons could be rendered outside the Dialog component actually